resources/pg_gem.rb

```
@@ -47,76 +58,39 @@
-      end
-    rescue Gem::Installer::ExtensionBuildError, Mixlib::ShellOut::ShellCommandFailed => e
-      build_essential 'for debian' do
-        compile_time true
-      end
+  build_essential 'essentially essential' do
+    compile_time true
+  end
 
-      # Are we an omnibus install?
-      raise if RbConfig.ruby.scan(/(chef|opscode)/).empty?
-      # Still here, must be omnibus. Lets make this thing install!
-      Chef::Log.warn 'Failed to properly build pg gem. Forcing properly linking and retrying (omnibus fix)'
-      gem_dir = e.message.scan(/will remain installed in ([^ ]+)/).flatten.first
-      raise unless gem_dir
-      gem_name = ::File.basename(gem_dir)
-      ext_dir = ::File.join(gem_dir, 'ext')
-      gem_exec = ::File.join(::File.dirname(RbConfig.ruby), 'gem')
-      new_content = <<-EOS
-require 'rbconfig'
-%w(
-configure_args
-LIBRUBYARG_SHARED
-LIBRUBYARG_STATIC
-LIBRUBYARG
-LDFLAGS
-).each do |key|
-  RbConfig::CONFIG[key].gsub!(/-Wl[^ ]+( ?\\/[^ ]+)?/, '')
-  RbConfig::MAKEFILE_CONFIG[key].gsub!(/-Wl[^ ]+( ?\\/[^ ]+)?/, '')
-end
-RbConfig::CONFIG['RPATHFLAG'] = ''
-RbConfig::MAKEFILE_CONFIG['RPATHFLAG'] = ''
-EOS
-      new_content << ::File.read(extconf_path = ::File.join(ext_dir, 'extconf.rb'))
-      ::File.open(extconf_path, 'w') do |file|
-        file.write(new_content)
-      end
-
-      lib_builder = execute 'generate pg gem Makefile' do
-        # [COOK-3490] pg gem install requires full path on RHEL
-
-        command "PATH=$PATH:/usr/pgsql-#{new_resource.client_version}/bin #{RbConfig.ruby} extconf.rb"
-        cwd ext_dir
-        action :nothing
-      end
-      lib_builder.run_action(:run)
-
-      lib_maker = execute 'make pg gem lib' do
-        command 'make'
-        cwd ext_dir
-        action :nothing
-      end
-      lib_maker.run_action(:run)
+  if ruby_version < 2.0
+    Chef::Log.fatal("pg gem requires a system Ruby installation of 2.0 or higher. \n Please install a global Ruby")
```

You want to raise here so you actually fail the run

---

I meant you should just raise "stuff be wrong!" instead of the chef log.